ethcore/sync/src/chain/mod.rs

```diff
@@ -649,24 +723,21 @@ impl ChainSync {
 	/// Resume downloading
 	fn continue_sync(&mut self, io: &mut SyncIo) {
 		// Collect active peers that can sync
-		let confirmed_peers: Vec<(PeerId, u8)> = self.peers.iter().filter_map(|(peer_id, peer)|
-			if peer.can_sync() {
+		let mut peers: Vec<(PeerId, u8)> = self.peers.iter().filter_map(|(peer_id, peer)|
+			if peer.can_sync() && peer.asking == PeerAsking::Nothing && self.active_peers.contains(peer_id) {
 				Some((*peer_id, peer.protocol_version))
 			} else {
 				None
 			}
 		).collect();
-		let mut peers: Vec<(PeerId, u8)> = confirmed_peers.iter().filter(|&&(peer_id, _)|
-			self.active_peers.contains(&peer_id)
-		).map(|v| *v).collect();
 
 		random::new().shuffle(&mut peers); //TODO: sort by rating
 		// prefer peers with higher protocol version
 		peers.sort_by(|&(_, ref v1), &(_, ref v2)| v1.cmp(v2));
```

Comment:
Don't we do some sort of peer sorting somewhere else as well? For connectivity issues, would it make sense to have a "global" peer sorting function that gets reused, that combines all the rules to find quality peers, like protocol version + connectivity + reputation kind of thing. Or do we literally only care about protocol version at this stage?

Reply:
I'm not quite sure about this. I think that the peers are only sorted here, so it would make sense to have some sort of sorting method. However I don't know if that's the right place, since we might have other useful data somewhere else, for example the first ping time or equivalent...